Fix linker flags for Python 3 on OSX

saraedum opened the pull request:
see https://github.com/pynac/pynac/issues/337 and https://github.com/ContinuumIO/anaconda-issues/issues/1761#issuecomment-361894754

saraedum committed:
Fix linker flags for Python 3 on OSX

see https://github.com/pynac/pynac/issues/337 and https://github.com/ContinuumIO/anaconda-issues/issues/1761#issuecomment-361894754

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe```) and found it was in an excellent condition.

saraedum commented:
@conda-forge-admin, please rerender

conda-forge-admin committed:
MNT: Re-rendered with conda-smithy 3.1.10 and pinning 2018.08.05

saraedum committed:
Special case OSX

LDSHARED is for building a shared library so these flags are too much. We also
want to build a .a and we do not want "-shared" in that case.
